db_management/db_upgrade.rst

```diff
@@ -0,0 +1,422 @@
+
+.. _db-upgrade/finalization:
+
+Finalization of the upgrade (ETA)
+---------------------------------
+
+Unfortunately, it is impossible to give time estimates for every upgrade request. Odoo offers so
+many possibilities (e.g. branding, workflows, customization, etc) that it can get tricky to upgrade,
+and translate to the new structure. If you use multiple apps managing sensitive data (e.g.,
+Accounting, Inventory, etc.), some cases may still require a human intervention, making the process
+slower.
+
+This is especially true during the first months following the release of a new major version, which
+can significantly lengthen the upgrade delay.
+
+In general, the ‘smaller’ the database, the quickest the upgrade. A single-user database that uses
+only CRM will be processed faster than a multi-company, multi-user database that uses Accounting,
+Sales, Purchase, and Manufacturing.
+
+So, in a nutshell, what can impact your upgrade lead time?
+
+* Source & targeted versions
+* Installed apps
+* Volume of data
+* Amount of customization (models, fields, methods, workflows, reports, website, etc.)
+* Installation of new apps or configuration changes after the start of the test phase
+
+Usually, the delays experienced during the first request (waiting time between the time you
+submitted your first request for a test upgrade) can generally give you an idea of the time to wait
+for the production request.
+
+.. _db-upgrade/custom-modules:
+
+Upgrade of the custom modules
+-----------------------------
+
+As stated in our :ref:`Odoo Enterprise Subscription Agreement, section "5.1 Standard charges"
+<charges_standard>`, this optional service is subject to additional fees.
+
+If you have a custom code, you can choose to have it upgraded by our services, by one of our
+`partners <https://www.odoo.com/partners>`_ or you can do it yourself.
+
+.. note::
+   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
+   our `Sales department <mailto:sales@odoo.com>`_.
+
+.. _db-upgrade/editions-change:
+
+Editions change (from Community to Enterprise)
+----------------------------------------------
+
+An upgrade does not cover a change of `Editions <https://www.odoo.com/page/editions>`_
+
+.. note::
+   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
+   our `Sales department <mailto:sales@odoo.com>`_.
+
+.. _db-upgrade/hosting-types-switch:
+
+Switching the hosting types (Self-hosted vs Online vs Odoo.sh)
+--------------------------------------------------------------
+
+An upgrade does not cover a change of `Hosting types <https://www.odoo.com/page/hosting-types>`_
+
+Open the following link to get :doc:`more information about how to change your hosting type
+<hosting_changes>`.
+
+.. note::
+   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
+   our `Sales department <mailto:sales@odoo.com>`_.
```

Comment:
Those notes are getting spammy. Is it really necessary to have so many of them?

Reply:
yes they are absolutely necessary; This documentation is based on the recurring requests we receive in the upgrade support pipe, because there is a lot of confusion for Odoo customers and employees about what the upgrade covers and what the processes are if it is related to a change of hosting or editions;
So yes please, keep them

Reply:
I understand but at the same time, that's a LOT of identical notes. It looks like advertising on some Parismatch blog post :stuck_out_tongue_closed_eyes:
In the end, I just make comments and suggestions here. Do what you want with them ;)

![unknown](https://user-images.githubusercontent.com/15896813/102357860-17706280-3faf-11eb-84c0-272e083f1acb.png)

Reply:
alright, we keep it, then :)
I replaced it in the code with substations

```
.. |assistance-contact| replace::
   If you need Odoo assistance on this matter, please contact your Odoo Account Manager or contact
   our `Sales department`_.
.. _Sales department: mailto:sales@odoo.com
```